easybuild/easyblocks/p/pdt.py

```diff
@@ -14,37 +14,44 @@
 
 @author Bernd Mohr (Juelich Supercomputing Centre)
 @author Markus Geimer (Juelich Supercomputing Centre)
+@author Alexander Grund (TU Dresden)
 """
 
 import os
 
 from easybuild.easyblocks.generic.configuremake import ConfigureMake
 import easybuild.tools.toolchain as toolchain
 from easybuild.tools.build_log import EasyBuildError
-from easybuild.tools.run import run_cmd
 
 
-class EB_PDT(ConfigureMake):
-    """Support for building/installing PDT."""
+def find_arch_dir(install_dir):
+    """Find architecture directory inside the install tree
 
-    def __init__(self, *args, **kwargs):
-        """Initialisation of custom class variables for PDT."""
-        super(EB_PDT, self).__init__(*args, **kwargs)
+    For simplicity any top-level folder containing a bin and lib directory is collected
+    Raises an error if multiple or none matching folders are found
+    Returns the architecture specific folder
+    """
+    arch_dirs = []
+    for entry in os.listdir(install_dir):
+        full_path = os.path.join(install_dir, entry)
+        # Only check directories which are not symlinks
+        # E.g. on x86_64 there are symlinks craycnl, mic_linux, ... to x86_64
+        if not os.path.isdir(full_path) or os.path.islink(full_path):
+            continue
+        bin_dir = os.path.join(full_path, 'bin')
+        lib_dir = os.path.join(full_path, 'lib')
+        if all(os.path.isdir(dir) for dir in (bin_dir, lib_dir)):
+            arch_dirs.append(full_path)
```

Comment:
bit shorter, since `bin_dir` or `lib_dir` are not used elsewhere:

```python
# only retain paths that both have a 'bin' and 'lib' subdirectory
if all(os.path.isdir(os.path.join(full_path, subdir)) for subdir in ['bin', 'lib']):

Reply:
Thanks. Could swear I tried that before but got over the line length. Probably didn't had the list comprehension back then